### PR TITLE
Log plugin supports different levels

### PR DESCRIPTION
**Pull request by cwahbong**
For #9. This is the new implementation based on rfc (#56).

---
This change is [Reviewable](https://reviewable.io/reviews/swiboe/swiboe/57)

### CODE REVIEW

File: src/plugin/log/base.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// in the project root for license information.
+use ::client;
+use ::rpc;
+use std::convert;
+use serde_json;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct Request {
+    pub message: String,
+    pub time: String,
```
Comment (SirVer) [inline src/plugin/log/base.rs]: String?

File: src/plugin/log/mod.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// Licensed under the Apache License, Version 2.0. See LICENSE.txt
+// in the project root for license information.
+use ::client;
+use ::error::Result;
+use ::plugin;
+use std::path;
+
+
+pub struct Plugin {
+    _client: client::Client,
```
Comment (SirVer) [inline src/plugin/log/mod.rs]: Not sure, but I think _ for private is uncommon for Rust.

File: src/plugin/log/mod.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+impl Plugin {
+    pub fn new(socket_name: &path::Path) -> Result<Self> {
+        let mut client = try!(client::Client::connect_unix(socket_name));
+        try!(plugin::register_rpc(&mut client, rpc_map! {
```
Comment (SirVer) [inline src/plugin/log/mod.rs]: rpc_map! is very elegant!!

**Comment by SirVer** [comment]
I think "time" is the open question here - not sure how to go about this.

File: src/plugin/log/mod.rs (hunk 2 above)
Comment (cwahbong) [inline src/plugin/log/mod.rs]: The _client is only for RAII. Rust compiler would give us a warning if the variable is not starts with an underscore.

File: src/plugin/log/base.rs (hunk 1 above)
Comment (cwahbong) [inline src/plugin/log/base.rs]: Oops. I think that a floating point would be better.

**Comment by SirVer** [comment]
thanks for clarifying. lgtm.

**Pull request merged**